gutils/fsys.c

```diff
@@ -1159,3 +1159,37 @@ char *GFileDirName(const char *path) {
     return GFileDirNameEx(path, 0);
 }
 
+char* GFileParent(char* path) {
+    GFile *gf_path = g_file_new_for_path(path);
+    GFile *gf_path_parent = g_file_get_parent(gf_path);
+
+    char* ret = g_file_get_path(gf_path_parent);
+
+    g_object_unref(gf_path);
+    g_object_unref(gf_path_parent);
+
+    return ret;
+}
+
+// If we have two files:
+// * /tmp/ex/test.sfd (file1)
+// * /tmp/ex/images/1.jpg (to_relative)
+//
+// The return is images/1.jpg
+char* GFileRelativize(char* file1, char* to_relative) {
+   GFile *gf_file1 = g_file_new_for_path(file1);
```

Comment:
It's a shame that `g_canonicalize_filename` is only available in relatively recent versions of glib, otherwise I'd suggest using that.

Also, I do see how relative paths can be useful (i.e. under version control), but I'm also not fully convinced it's the right thing to do, especially when it only does this if the sfd is already saved. It just seems much too fragile, and a bit too 'magic' overall.

Reply:
Ok yeah this definitely needs more thought. 

Is it really worth it to write out relative paths?

This breaks as soon as you do a 'save as' to a different folder, assuming you've already saved it in some other location beforehand.

Reply:
Absolutely, absolutely it is worth it. Without relative paths, no one else can use my SFDs. It's a requirement for this feature to be useful.

Reply:
I totally agree with @ctrlcctrlv on this. You ought to be able to move your project directory around on the computer or to transmit it to others without it breaking. I sometimes need to manually edit Xournal files in order to make paths relative, and then it changes them back every time I edit them, which is very irritating.